### PR TITLE
Feature/libvirt iscsi

### PR DESCRIPTION
**Pull request by arbulu89**
Add iscsi support to libvirt.
Now, using the `shared_storage_type` variable in `tfvars` we can choose between `iscsi` and `shared-disk`

### CODE REVIEW

File: salt/pillar/iscsi_initiator.sls [hunk 1]
```diff
@@ -1,12 +0,0 @@
-iscsi:
```
Comment (diegoakechi) [inline salt/pillar/iscsi_initiator.sls]: Is this file being dropped intentionally?

Comment (arbulu89) [reply salt/pillar/iscsi_initiator.sls]: Yes. The truth is that we were not using it. 
Currently, we don't use the iscsi formula to setup the initiator as we found issues there.
In the future, we should check if it worth to use the formula or not.

File: libvirt/terraform/terraform.tfvars.example [hunk 2]
```diff
@@ -6,6 +6,12 @@ iprange = "192.168.XXX.Y/24"
 host_ips = ["192.168.XXX.Y", "192.168.XXX.Y+1"]
 additional_repos = {}
 
+# Shared storage type information
+shared_storage_type = "iscsi"
+iscsi_srv_ip = "192.168.XXX.Y+2"
+iscsi_image = "url-to-your-sles4sap-image" # sles15 or above
+
+
```
Comment (ldevulder) [inline libvirt/terraform/terraform.tfvars.example]: One blank line should be sufficient :-)

File: salt/default/minimal.sls [hunk 3]
```diff
@@ -1,6 +1,5 @@
 include:
   - default.setup
-  - default.ip_workaround
```
Comment (ldevulder) [inline salt/default/minimal.sls]: Why is it not needed anymore?

Comment (arbulu89) [reply salt/default/minimal.sls]: It still needed. Only moved to other folder to make it more generic for libvirt, as all of the machines created in this provider has the same issue (ip address not assigned to eth1).

**Comment by ldevulder** [comment]
@arbulu89 I had no issue on my tests, but please next time if we said that we want to test, first wait for our go to merge :-) (we can revert in case of issue, but it's sometimes not so easy).

**Comment by arbulu89** [comment]
> @arbulu89 I had no issue on my tests, but please next time if we said that we want to test, first wait for our go to merge :-) (we can revert in case of issue, but it's sometimes not so easy).

I'm really sorry @ldevulder . I just misunderstood your last sentence...
Tell me if you find something that doesn't work properly. I will restore back the master branch in that case

**Comment by ldevulder** [comment]
@arbulu89 no issue, that fine!

And yes I re-read my sentence and it was not clear enough... My English was bad indeed... sorry!

**Pull request merged**